packages/sit-onyx/src/composables/useLink.ts

```diff
@@ -0,0 +1,74 @@
+import { inject, type InjectionKey } from "vue";
+import type { LinkTarget } from "../components/OnyxRouterLink/types";
+import { isExternalLink } from "../utils";
+
+/**
+ * Internal behavior of the `OnyxRouterLink`. Should be used for navigation. Considers the user provided router.
+ */
+export const useLink = () => {
+  const router = inject(ROUTER_INJECTION_KEY, undefined);
+
+  /**
+   * Navigates to the given URL by considering the user provided router if the link is internal.
+   *
+   * @param e Mouse event of a `<a>` element, e.g. from a click listener.
+   * @example
+   * ```vue
+   * <script lang="ts" setup>
+   * const { navigate } = useLink();
+   * </script>
+   *
+   * <template>
+   *   <a
+   *     :href="props.href"
+   *     :target="props.target"
+   *     @click="navigate($event, props.href, props.target)"
+   *   >
+   *     <slot></slot>
+   *   </a>
+   * </template>
+   * ```
+   */
+  const navigate = (e: MouseEvent, href: string, target: LinkTarget) => {
+    if (!shouldHandleNavigation(e)) return;
+    const isExternal = isExternalLink(href);
+    if (isExternal || !router) window.open(href, target);
+    else router.push(href);
+  };
```

Comment:
IMHO we should only handle the link when a router is given and it's internal.
Or rather `window.open` should be left to the default browser behavior.

External links can be handled by the browser, the the user has to set the target correctly.
But we could consider to set the target on the `a` accordingly for external links, when left undefined by the user.

If there is a reason that I am overseeing, we might need an explanatory comment in the code.

```suggestion
  const navigate = (e: MouseEvent, href: string) => {
    if (shouldHandleNavigation(e) && !isExternalLink(href) && router) {
      router.push(href);
    }
  };
```